chore(deps) Updated packages for security vulnerabilities

sarvex opened the pull request:
[Describe your pull request here. Please read the text below the line, and make sure you follow the checklist.]

* * *

## Pull request checklist

Read the [Contribution Guidelines](https://github.com/nlohmann/json/blob/develop/.github/CONTRIBUTING.md) for detailed information.

- [x]  Changes are described in the pull request, or an [existing issue is referenced](https://github.com/nlohmann/json/issues).
- [ ]  The test suite [compiles and runs](https://github.com/nlohmann/json/blob/develop/README.md#execute-unit-tests) without error.
- [ ]  [Code coverage](https://coveralls.io/github/nlohmann/json) is 100%. Test cases can be added by editing the [test suite](https://github.com/nlohmann/json/tree/develop/test/src).
- [ ]  The source code is amalgamated; that is, after making changes to the sources in the `include/nlohmann` directory, run `make amalgamate` to create the single-header files `single_include/nlohmann/json.hpp` and `single_include/nlohmann/json_fwd.hpp`. The whole process is described [here](https://github.com/nlohmann/json/blob/develop/.github/CONTRIBUTING.md#files-to-change).

## Please don't

- The C++11 support varies between different **compilers** and versions. Please note the [list of supported compilers](https://github.com/nlohmann/json/blob/master/README.md#supported-compilers). Some compilers like GCC 4.7 (and earlier), Clang 3.3 (and earlier), or Microsoft Visual Studio 13.0 and earlier are known not to work due to missing or incomplete C++11 support. Please refrain from proposing changes that work around these compiler's limitations with `#ifdef`s or other means.
- Specifically, I am aware of compilation problems with **Microsoft Visual Studio** (there even is an [issue label](https://github.com/nlohmann/json/issues?utf8=✓&q=label%3A%22visual+studio%22+) for this kind of bug). I understand that even in 2016, complete C++11 support isn't there yet. But please also understand that I do not want to drop features or uglify the code just to make Microsoft's sub-standard compiler happy. The past has shown that there are ways to express the functionality such that the code compiles with the most recent MSVC - unfortunately, this is not the main objective of the project.
- Please refrain from proposing changes that would **break [JSON](https://json.org) conformance**. If you propose a conformant extension of JSON to be supported by the library, please motivate this extension.
- Please do not open pull requests that address **multiple issues**.

dependabot[bot] committed:
:arrow_up: Bump pymdown-extensions from 9.9 to 10.0 in /docs/mkdocs

Bumps [pymdown-extensions](https://github.com/facelessuser/pymdown-extensions) from 9.9 to 10.0.
- [Release notes](https://github.com/facelessuser/pymdown-extensions/releases)
- [Commits](https://github.com/facelessuser/pymdown-extensions/compare/9.9...10.0)

---
updated-dependencies:
- dependency-name: pymdown-extensions
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

sarvex committed:
Merge pull request #5 from sarvex/dependabot/pip/docs/mkdocs/pymdown-extensions-10.0

⬆️ Bump pymdown-extensions from 9.9 to 10.0 in /docs/mkdocs

dependabot[bot] committed:
:arrow_up: Bump gitpython from 3.1.29 to 3.1.30 in /docs/mkdocs

Bumps [gitpython](https://github.com/gitpython-developers/GitPython) from 3.1.29 to 3.1.30.
- [Release notes](https://github.com/gitpython-developers/GitPython/releases)
- [Changelog](https://github.com/gitpython-developers/GitPython/blob/main/CHANGES)
- [Commits](https://github.com/gitpython-developers/GitPython/compare/3.1.29...3.1.30)

---
updated-dependencies:
- dependency-name: gitpython
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

sarvex committed:
Create renovate.json

sarvex committed:
Merge pull request #6 from sarvex/dependabot/pip/docs/mkdocs/gitpython-3.1.30

⬆️ Bump gitpython from 3.1.29 to 3.1.30 in /docs/mkdocs

sarvex committed:
Create codeql.yml

dependabot[bot] committed:
:arrow_up: Bump requests from 2.28.1 to 2.31.0 in /docs/mkdocs

Bumps [requests](https://github.com/psf/requests) from 2.28.1 to 2.31.0.
- [Release notes](https://github.com/psf/requests/releases)
- [Changelog](https://github.com/psf/requests/blob/main/HISTORY.md)
- [Commits](https://github.com/psf/requests/compare/v2.28.1...v2.31.0)

---
updated-dependencies:
- dependency-name: requests
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

sarvex committed:
Merge pull request #7 from sarvex/dependabot/pip/docs/mkdocs/requests-2.31.0

⬆️ Bump requests from 2.28.1 to 2.31.0 in /docs/mkdocs

sarvex committed:
Merge branch 'nlohmann:develop' into main

mbeaulie commented:
I believe nltk needs to be bumped 3.8 -> 3.8.1 for VS-2022-0437 and VS-2022-0438 security fixed too.

dependabot[bot] committed:
:arrow_up: Bump tornado from 6.2 to 6.3.2 in /docs/mkdocs

Bumps [tornado](https://github.com/tornadoweb/tornado) from 6.2 to 6.3.2.
- [Changelog](https://github.com/tornadoweb/tornado/blob/master/docs/releases.rst)
- [Commits](https://github.com/tornadoweb/tornado/compare/v6.2.0...v6.3.2)

---
updated-dependencies:
- dependency-name: tornado
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

sarvex committed:
Merge pull request #8 from sarvex/dependabot/pip/docs/mkdocs/tornado-6.3.2

⬆️ Bump tornado from 6.2 to 6.3.2 in /docs/mkdocs

sarvex committed:
Merge branch 'nlohmann:develop' into main

dependabot[bot] committed:
:arrow_up: Bump pygments from 2.13.0 to 2.15.0 in /docs/mkdocs

Bumps [pygments](https://github.com/pygments/pygments) from 2.13.0 to 2.15.0.
- [Release notes](https://github.com/pygments/pygments/releases)
- [Changelog](https://github.com/pygments/pygments/blob/master/CHANGES)
- [Commits](https://github.com/pygments/pygments/compare/2.13.0...2.15.0)

---
updated-dependencies:
- dependency-name: pygments
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
:arrow_up: Bump certifi from 2022.12.7 to 2023.7.22 in /docs/mkdocs

Bumps [certifi](https://github.com/certifi/python-certifi) from 2022.12.7 to 2023.7.22.
- [Commits](https://github.com/certifi/python-certifi/compare/2022.12.07...2023.07.22)

---
updated-dependencies:
- dependency-name: certifi
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
:arrow_up: Bump gitpython from 3.1.30 to 3.1.32 in /docs/mkdocs

Bumps [gitpython](https://github.com/gitpython-developers/GitPython) from 3.1.30 to 3.1.32.
- [Release notes](https://github.com/gitpython-developers/GitPython/releases)
- [Changelog](https://github.com/gitpython-developers/GitPython/blob/main/CHANGES)
- [Commits](https://github.com/gitpython-developers/GitPython/compare/3.1.30...3.1.32)

---
updated-dependencies:
- dependency-name: gitpython
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

sarvex committed:
Merge pull request #11 from sarvex/dependabot/pip/docs/mkdocs/gitpython-3.1.32

⬆️ Bump gitpython from 3.1.30 to 3.1.32 in /docs/mkdocs

dependabot[bot] committed:
:arrow_up: Bump tornado from 6.3.2 to 6.3.3 in /docs/mkdocs

Bumps [tornado](https://github.com/tornadoweb/tornado) from 6.3.2 to 6.3.3.
- [Changelog](https://github.com/tornadoweb/tornado/blob/master/docs/releases.rst)
- [Commits](https://github.com/tornadoweb/tornado/compare/v6.3.2...v6.3.3)

---
updated-dependencies:
- dependency-name: tornado
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

sarvex committed:
Merge pull request #12 from sarvex/dependabot/pip/docs/mkdocs/tornado-6.3.3

⬆️ Bump tornado from 6.3.2 to 6.3.3 in /docs/mkdocs

sarvex committed:
Merge pull request #10 from sarvex/dependabot/pip/docs/mkdocs/certifi-2023.7.22

⬆️ Bump certifi from 2022.12.7 to 2023.7.22 in /docs/mkdocs

sarvex committed:
Merge pull request #9 from sarvex/dependabot/pip/docs/mkdocs/pygments-2.15.0

⬆️ Bump pygments from 2.13.0 to 2.15.0 in /docs/mkdocs

sarvex committed:
Merge branch 'nlohmann:develop' into main

nlohmann commented:
We only use Python for generating the documentation pages. I don't see the value of adding this.

dependabot[bot] committed:
Bump gitpython from 3.1.32 to 3.1.34 in /docs/mkdocs

Bumps [gitpython](https://github.com/gitpython-developers/GitPython) from 3.1.32 to 3.1.34.
- [Release notes](https://github.com/gitpython-developers/GitPython/releases)
- [Changelog](https://github.com/gitpython-developers/GitPython/blob/main/CHANGES)
- [Commits](https://github.com/gitpython-developers/GitPython/compare/3.1.32...3.1.34)

---
updated-dependencies:
- dependency-name: gitpython
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

sarvex committed:
Merge pull request #13 from sarvex/dependabot/pip/docs/mkdocs/gitpython-3.1.34

Bump gitpython from 3.1.32 to 3.1.34 in /docs/mkdocs

dependabot[bot] committed:
Bump gitpython from 3.1.34 to 3.1.35 in /docs/mkdocs

Bumps [gitpython](https://github.com/gitpython-developers/GitPython) from 3.1.34 to 3.1.35.
- [Release notes](https://github.com/gitpython-developers/GitPython/releases)
- [Changelog](https://github.com/gitpython-developers/GitPython/blob/main/CHANGES)
- [Commits](https://github.com/gitpython-developers/GitPython/compare/3.1.34...3.1.35)

---
updated-dependencies:
- dependency-name: gitpython
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

sarvex committed:
Merge pull request #14 from sarvex/dependabot/pip/docs/mkdocs/gitpython-3.1.35

Bump gitpython from 3.1.34 to 3.1.35 in /docs/mkdocs

dependabot[bot] committed:
Bump urllib3 from 1.26.13 to 1.26.17 in /docs/mkdocs

Bumps [urllib3](https://github.com/urllib3/urllib3) from 1.26.13 to 1.26.17.
- [Release notes](https://github.com/urllib3/urllib3/releases)
- [Changelog](https://github.com/urllib3/urllib3/blob/main/CHANGES.rst)
- [Commits](https://github.com/urllib3/urllib3/compare/1.26.13...1.26.17)

---
updated-dependencies:
- dependency-name: urllib3
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

sarvex committed:
Merge pull request #15 from sarvex/dependabot/pip/docs/mkdocs/urllib3-1.26.17

Bump urllib3 from 1.26.13 to 1.26.17 in /docs/mkdocs

sarvex committed:
Merge branch 'nlohmann:develop' into main

dependabot[bot] committed:
Bump gitpython from 3.1.35 to 3.1.37 in /docs/mkdocs

Bumps [gitpython](https://github.com/gitpython-developers/GitPython) from 3.1.35 to 3.1.37.
- [Release notes](https://github.com/gitpython-developers/GitPython/releases)
- [Changelog](https://github.com/gitpython-developers/GitPython/blob/main/CHANGES)
- [Commits](https://github.com/gitpython-developers/GitPython/compare/3.1.35...3.1.37)

---
updated-dependencies:
- dependency-name: gitpython
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

sarvex committed:
Merge pull request #16 from sarvex/dependabot/pip/docs/mkdocs/gitpython-3.1.37

Bump gitpython from 3.1.35 to 3.1.37 in /docs/mkdocs

dependabot[bot] committed:
Bump urllib3 from 1.26.17 to 1.26.18 in /docs/mkdocs

Bumps [urllib3](https://github.com/urllib3/urllib3) from 1.26.17 to 1.26.18.
- [Release notes](https://github.com/urllib3/urllib3/releases)
- [Changelog](https://github.com/urllib3/urllib3/blob/main/CHANGES.rst)
- [Commits](https://github.com/urllib3/urllib3/compare/1.26.17...1.26.18)

---
updated-dependencies:
- dependency-name: urllib3
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

sarvex committed:
Merge pull request #17 from sarvex/dependabot/pip/docs/mkdocs/urllib3-1.26.18

Bump urllib3 from 1.26.17 to 1.26.18 in /docs/mkdocs

sarvex committed:
Merge branch 'nlohmann:develop' into main

renovate[bot] committed:
Update dependency GitPython to v3.1.40

sarvex committed:
Merge pull request #19 from sarvex/renovate/gitpython-3.x

Update dependency GitPython to v3.1.40

renovate[bot] committed:
Update actions/github-script action to v6.4.1

renovate[bot] committed:
Update dependency PyYAML to v6.0.1

sarvex committed:
Merge pull request #18 from sarvex/renovate/actions-github-script-6.x

Update actions/github-script action to v6.4.1

sarvex committed:
Merge pull request #22 from sarvex/renovate/pyyaml-6.x

Update dependency PyYAML to v6.0.1

renovate[bot] committed:
Update dependency MarkupSafe to v2.1.3

sarvex committed:
Merge pull request #21 from sarvex/renovate/markupsafe-2.x

Update dependency MarkupSafe to v2.1.3

renovate[bot] committed:
Update dependency mkdocs-redirects to v1.2.1

sarvex committed:
Merge pull request #26 from sarvex/renovate/mkdocs-redirects-1.x

Update dependency mkdocs-redirects to v1.2.1

renovate[bot] committed:
Update dependency gitdb to v4.0.11

renovate[bot] committed:
Update dependency pytz to v2022.7.1

sarvex committed:
Merge pull request #24 from sarvex/renovate/gitdb-4.x

Update dependency gitdb to v4.0.11

sarvex committed:
Merge pull request #28 from sarvex/renovate/pytz-2022.x

Update dependency pytz to v2022.7.1

renovate[bot] committed:
Update dependency nltk to v3.8.1

sarvex committed:
Merge pull request #27 from sarvex/renovate/nltk-3.x

Update dependency nltk to v3.8.1

renovate[bot] committed:
Update dependency markdown-include to v0.8.1

sarvex committed:
Merge pull request #25 from sarvex/renovate/markdown-include-0.x

Update dependency markdown-include to v0.8.1

renovate[bot] committed:
Update dependency smmap to v5.0.1

renovate[bot] committed:
Update dependency Babel to v2.13.1

sarvex committed:
Merge pull request #30 from sarvex/renovate/babel-2.x

Update dependency Babel to v2.13.1

sarvex committed:
Merge pull request #29 from sarvex/renovate/smmap-5.x

Update dependency smmap to v5.0.1

renovate[bot] committed:
Update dependency click to v8.1.7

sarvex committed:
Merge pull request #23 from sarvex/renovate/click-8.x

Update dependency click to v8.1.7

dependabot[bot] committed:
Bump gitpython from 3.1.40 to 3.1.41 in /docs/mkdocs

Bumps [gitpython](https://github.com/gitpython-developers/GitPython) from 3.1.40 to 3.1.41.
- [Release notes](https://github.com/gitpython-developers/GitPython/releases)
- [Changelog](https://github.com/gitpython-developers/GitPython/blob/main/CHANGES)
- [Commits](https://github.com/gitpython-developers/GitPython/compare/3.1.40...3.1.41)

---
updated-dependencies:
- dependency-name: gitpython
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump jinja2 from 3.1.2 to 3.1.3 in /docs/mkdocs

Bumps [jinja2](https://github.com/pallets/jinja) from 3.1.2 to 3.1.3.
- [Release notes](https://github.com/pallets/jinja/releases)
- [Changelog](https://github.com/pallets/jinja/blob/main/CHANGES.rst)
- [Commits](https://github.com/pallets/jinja/compare/3.1.2...3.1.3)

---
updated-dependencies:
- dependency-name: jinja2
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

sarvex committed:
Merge pull request #38 from sarvex/dependabot/pip/docs/mkdocs/jinja2-3.1.3

Bump jinja2 from 3.1.2 to 3.1.3 in /docs/mkdocs

sarvex committed:
Merge pull request #37 from sarvex/dependabot/pip/docs/mkdocs/gitpython-3.1.41

Bump gitpython from 3.1.40 to 3.1.41 in /docs/mkdocs
